roomserver/internal/input/input.go

```diff
@@ -379,7 +379,7 @@ func (w *worker) _next() {
 		}
 		errString = err.Error()
 	} else {
-		if err := msg.AckSync(); err != nil {
+		if err = msg.AckSync(); err != nil {
```

Comment:
It doesn't matter too much, but `if err:=...` is a very common Go pattern that will indeed shadow previously declared errors, but that seems to be the convention.

Reply:
I agree - the value of just recognizing the pattern (`err :=`) seems to outweigh the almost 0% chance of the shadowing causing any unforeseen things. I don't know why the checks here are so nitpicky